torch/distributed/_tensor/ops/experimental_ops.py

```diff
@@ -88,3 +93,51 @@ def nll_loss_backward_rules(op_schema: OpSchema) -> OutputSharding:
     input_spec = op_schema.args_schema[1]
     assert isinstance(input_spec, DTensorSpec)
     return OutputSharding(input_spec)
+
+
+@register_prop_rule(aten.native_layer_norm_backward.default)
+def _prop_native_layer_norm_backward(op_schema: OpSchema) -> OutputSharding:
+    (
+        grad,
+        input,
+        normalized_shape,
+        result1,
+        result2,
+        weight,
+        bias,
+        grad_input_mask,
+    ) = op_schema.args_schema
+    assert isinstance(grad, DTensorSpec)
+    assert isinstance(grad_input_mask, (list, tuple))
+    if weight is not None:
+        assert isinstance(weight, DTensorSpec)
+        assert all(isinstance(s, Replicate) for s in weight.placements)
+    if bias is not None:
+        assert isinstance(bias, DTensorSpec)
+        assert all(isinstance(s, Replicate) for s in bias.placements)
+
+    weight_grad = (
+        DTensorSpec(
+            mesh=weight.mesh,
+            placements=tuple([_Partial()] * weight.mesh.ndim),
```

Comment:
curious how are you triggering the backward allreduce for the gradients now? since the param.grads are partial tensors, it's by default lazy staying as partial dtensor, if we use dtensor alone the allreduce will automatically be triggered in the optimizer.step. But if we combine with FlatParam FSDP together we would need to eagerly trigger this once the gradient is ready (i.e. via register_hook on parameters)

Reply:
Ideally we don't want to overwrite the placement always as Partial() or any "hardcoded" placements, instead, we should rely on the sharing propagation rule to make the decision. Specifically for the DTensor-based experimental WHEN model, the "overwrite" operation is unnecessary. However, to maintain consistency with the current behavior here: https://github.com/pytorch/pytorch/blob/main/torch/distributed/_spmd/experimental_ops.py#L290 I introduced these lines as well.

If we have concerns, I can remove them.